product/model.py

```diff
@@ -396,6 +399,13 @@ def image_reference(self):
     def image_name(self):
         return self.raw.get('name')
 
+    @functools.lru_cache
+    def content_addressable_id(self):
```

Comment:
not sure I like the name. how about `digest_image_reference`?
or `image_reference_with_digest`
or maybe adding a parameter to `image_reference` (e.g tag='symbolic'|'digest')
in the latter case, only the `image_digest` function would require caching